src/bit-systems/audio-zone-system.ts

```diff
@@ -0,0 +1,325 @@
+
+// We apply the most restrictive audio parameters
+const paramsReducer = (acc: Partial<AudioSettings>, curr: Partial<AudioSettings>): Partial<AudioSettings> => {
+  if (!curr && !acc) return {} as AudioSettings;
+  else if (curr && !acc) return curr;
+  else if (!curr && acc) return acc;
+  else
+    return REDUCED_KEYS.reduce(
+      (result: ReducedAudioSettings, key: ReducedAudioSettingsKeys): Partial<AudioSettings> => {
+        if (curr[key] !== undefined && acc[key] !== undefined) {
+          result[key] = Math.min(acc[key]!, curr[key]!);
+        } else if (curr[key] !== undefined && acc[key] === undefined) {
+          result[key] = curr[key];
+        } else if (curr[key] === undefined && acc[key] !== undefined) {
+          result[key] = acc[key];
+        }
+        return result;
+      },
+      {}
+    );
+};
+
+const addOrRemoveZone = (zones: Set<number>, zoneEid: number, position: Vector3) => {
+  if (!zones) return;
+  const isInZone = isZoneEnabled(zoneEid) && zoneContains(zoneEid, position);
+  const wasInZone = zones.has(zoneEid);
+  if (isInZone && !wasInZone) {
+    zones.add(zoneEid);
+  } else if (!isInZone && wasInZone) {
+    zones.delete(zoneEid);
+  }
+};
+
+const clearEntity = (entityEid: number) => {
+  const zones = prevZones.get(entityEid);
+  if (zones) {
+    zones.clear();
+    currZones.get(entityEid)?.forEach(zone => zones.add(zone));
+  }
+};
+
+const currZones = new Map<number, Set<number>>();
+const prevZones = new Map<number, Set<number>>();
+const aabbs = new Map<number, Box3>();
+
+const listenerPos = new Vector3();
+const audioZoneQuery = defineQuery([AudioZone]);
+const audioZoneEnterQuery = enterQuery(audioZoneQuery);
+const audioZoneExitQuery = exitQuery(audioZoneQuery);
+const audioEmitterQuery = defineQuery([AudioEmitter]);
+const audioEmitterEnterQuery = enterQuery(audioEmitterQuery);
+const audioEmitterExitQuery = exitQuery(audioEmitterQuery);
+
+export function audioZoneSystem(world: HubsWorld) {
+  audioEmitterEnterQuery(world).forEach(entityEid => {
+    currZones.set(entityEid, new Set());
+    prevZones.set(entityEid, new Set());
+  });
+  currZones.set(APP.audioListener.eid!, new Set());
+  audioEmitterExitQuery(world).forEach(entityEid => {
+    currZones.delete(entityEid);
+    prevZones.delete(entityEid);
+  });
+
+  audioZoneEnterQuery(world).forEach(zoneEid => {
+    const obj = APP.world.eid2obj.get(zoneEid)!;
+    const aabb = new Box3();
+    aabb.setFromObject(obj);
+    aabbs.set(zoneEid, aabb);
+
+    const isDebugEnabled = APP.store.state.preferences.showAudioDebugPanel;
+    isDebugEnabled && !debugObjects.has(zoneEid) && addZoneDebugObject(APP.world, zoneEid);
+  });
+  audioZoneExitQuery(world).forEach(zoneEid => {
+    aabbs.delete(zoneEid);
+    debugObjects.delete(zoneEid);
+
+    const isDebugEnabled = APP.store.state.preferences.showAudioDebugPanel;
+    isDebugEnabled && debugObjects.has(zoneEid) && releaseZoneDebugObject(APP.world, zoneEid);
```

Comment:
It seems like these lines belong in the audio-debug-system

Reply:
This are in charge of showing/hiding the audio zone debug objects based on the debug panel preference status when an audio zone is added/removed. In this case we can't rely on an preference status update event, we need to explicitly check.